packages/swarm-components/src/TextInput.jsx

```diff
@@ -87,6 +96,7 @@ export const TextInput = (props: Props): React$Element<*> => {
           <Icon shape={iconShape} size={iconSize} aria-hidden={true} />
         </span>
       )}
+      {maxLength && <CharCount>{remainingCharacters}</CharCount>}
```

Comment:
the interface for `CharCount` seems a bit awkward since the parent will likely need `getRemainingCharacters` in most cases - why not set up a props interface that takes `maxLength` and `currentLength` props and determines what to do from there? That would also allow us to try more sophisticated behavior, e.g. where the character count is only displayed when there are less than 50 characters remaining (or current length is 90% of maxLength) or something.

I could see an argument for why the parent should determine when the count is or isn't displayed, but that also allows deviation from a standard and mixes rendering responsibilities

Reply:
I was just using the current implementation with `Textarea`. @hsbacot thoughts?

Reply:
Either is fine. `<CharCount />` shouldn't be exported from swarm so that should allow for more sophisticated behavior to be encapsulated in `Textarea` or `TextInput`. I think that's something we should solve for if & when that becomes a requirement.

Reply:
I updated the interface to the `CharCount` component. We still want to export `getRemainingCharCount` because `Textarea` needs it to determine error states for the text area itself, however, this functionality is not working even in the current version. Might want to fix that as a separate defect. Also, we'd probably want to have the same logic/interface for TextInput, which we currently don't.